scripts/fuzzing.sh

```diff
@@ -3,16 +3,32 @@ set -e
 source ./scripts/test_lib.sh
 
 GO_CMD="go"
-fuzz_time=${FUZZ_TIME:-"300s"}
-target_path=${TARGET_PATH:-"./server/etcdserver/api/v3rpc"}
-TARGETS="FuzzTxnRangeRequest  FuzzTxnPutRequest  FuzzTxnDeleteRangeRequest"
+fuzz_time=${FUZZ_TIME:-"3s"}
 
+function run_fuzzer() {
+    target_path=$1
+    target=$2
 
-for target in ${TARGETS}; do
     log_callout -e "\\nExecuting fuzzing with target ${target} in $target_path with a timeout of $fuzz_time\\n"
     run pushd "${target_path}"
         $GO_CMD test -fuzz "${target}" -fuzztime "${fuzz_time}"
     run popd
     log_success -e "\\COMPLETED: fuzzing with target $target in $target_path \\n"
+}
+
+# api fuzzers
+target_path=${TARGET_PATH:-"./server/etcdserver/api/v3rpc"}
+TARGETS="FuzzTxnRangeRequest  FuzzTxnPutRequest  FuzzTxnDeleteRangeRequest"
+
+
+for target in ${TARGETS}; do
+    run_fuzzer "${target_path}" "${target}"
 done
 
+# raft fuzzers
+target_path=${TARGET_PATH:-"./raft"}
+TARGETS="FuzzStep"
+
+for target in ${TARGETS}; do
+    run_fuzzer "${target_path}" "${target}"
+done
```

Comment:
If users specify the `TARGET_PATH`, then this script will run into issue? 

I would suggest some script/code something like below so as to make it more extensible,
```
for target_path in ${TARGET_PATHS}; do 
    for target in ${TARGETS}; do
        run_fuzzer "${target_path}" "${target}"
    done
done
```

Reply:
I wonder if that is an issue. This comes from the current version of `scripts/fuzzing.sh`:
https://github.com/etcd-io/etcd/blob/2f558ca0dbf1217c1cc5b82a1e2aec428bf6a04f/scripts/fuzzing.sh#L1-L8

Is `TARGET_PATH` meant to be specified by users?

Reply:
When we want to manually run part of the fuzzing test, then we might set the `TARGET_PATH`.

Reply:
Given that each target path will have differently named fuzz tests, I think it would look something like this. Is that what you had in mind? 
```bash
#!/usr/bin/env bash
set -e
source ./scripts/test_lib.sh

GO_CMD="go"
fuzz_time=${FUZZ_TIME:-"3s"}

function run_fuzzer() {
    target_path=$1
    target=$2

    log_callout -e "\\nExecuting fuzzing with target ${target} in $target_path with a timeout of $fuzz_time\\n"
    run pushd "${target_path}"
        $GO_CMD test -fuzz "${target}" -fuzztime "${fuzz_time}"
    run popd
    log_success -e "\\COMPLETED: fuzzing with target $target in $target_path \\n"
}

if [ ! -z "${TARGET_PATHS}" ] # If TARGET_PATHS is set
then
        for target_path in "${TARGET_PATHS}"; do
                if [ $target_path == "./server/etcdserver/api/v3rpc" ]; then
                        TARGETS="FuzzTxnRangeRequest  FuzzTxnPutRequest  FuzzTxnDeleteRangeRequest"
                        for target in ${TARGETS}; do
                                run_fuzzer "${target_path}" "${target}"
                        done
                elif [ $target_path == "./raft" ]; then
                        TARGETS="FuzzStep"
                        for target in ${TARGETS}; do
                                run_fuzzer "${target_path}" "${target}"
                        done
                fi
        done
else
        echo "TARGET_PATHS not set. Running all fuzzers"
        # api fuzzers
        target_path=${TARGET_PATH:-"./server/etcdserver/api/v3rpc"}
        TARGETS="FuzzTxnRangeRequest  FuzzTxnPutRequest  FuzzTxnDeleteRangeRequest"

        for target in ${TARGETS}; do
                run_fuzzer "${target_path}" "${target}"
        done

        # raft fuzzers
        target_path=${TARGET_PATH:-"./raft"}
        TARGETS="FuzzStep"

        for target in ${TARGETS}; do
                run_fuzzer "${target_path}" "${target}"
        done
fi
```

Reply:
Yes, when $TARGET_PATHS isn't set. Can we use a two level loops just I mentioned in https://github.com/etcd-io/etcd/pull/14674#discussion_r1015821330? Otherwise, when any new target is added, then the new contributor needs to copy-paste the duplicated code.